src/Adapt/Adapt.component.tsx

```diff
@@ -0,0 +1,101 @@
+// VENDOR
+import * as React from 'react';
+import classNames from 'classnames';
+import styled, {
+    up,
+    css,
+    down,
+    ThemeContext,
+} from '@xstyled/styled-components';
+import { layout, LayoutProps, space, SpaceProps } from '@xstyled/system';
+
+const { useContext } = React;
+
+interface ForEach {
+    [k: string]: any;
+}
+
+interface AdaptProps
+    extends React.HTMLAttributes<HTMLDivElement>,
+        LayoutProps,
+        SpaceProps {
+    className?: string;
+    forEach: ForEach;
+    children: (value: any, breakpoint: string) => any;
```

Comment:
Won't this always need to return `JSX` or `null`?

Reply:
I imagined the scenario that someone just passes in a string- but you're right, that wouldn't work with this system (need a dom node to target).

The possibility of `null` possibility is also very interesting. May need to deal with that specifically

Reply:
I think `any` is prolly fine for now, but whenever I see that in TS I assume it's incorrect.